File: src/main/java/net/treset/minecraftlauncher/util/ui/sort/InstanceDetailsTimeComparator.java
package net.treset.minecraftlauncher.util.ui.sort;

import net.treset.minecraftlauncher.LauncherApplication;
import net.treset.minecraftlauncher.data.InstanceData;

import java.util.Comparator;

public class InstanceDetailsTimeComparator implements Comparator<InstanceData> {
    @Override
    public int compare(InstanceData o1, InstanceData o2) {
        return (int) (o2.getInstance().getValue().getTotalTime() - o1.getInstance().getValue().getTotalTime());
    }

    @Override
    public String toString() {
        return LauncherApplication.stringLocalizer.get("sort.time");
    }
}

File: client/composeApp/src/commonMain/kotlin/net/treset/treelauncher/backend/util/sort/InstanceDetailsTimeComparator.kt
package net.treset.treelauncher.backend.util.sort

import net.treset.treelauncher.backend.data.InstanceData
import net.treset.treelauncher.localization.strings

class InstanceDetailsTimeComparator : Comparator<InstanceData> {
    override fun compare(o1: InstanceData, o2: InstanceData): Int {
        return (o2.instance.second.totalTime - o1.instance.second.totalTime).toInt()
    }

    override fun toString(): String = strings().sorts.time()
}
